### ISSUE TITLE
Don't sort contents of a data class / data model

### ISSUE THREAD

**Issue by jamesrwelch** [MauroDataMapper/mdm-core#268]
Sometimes, it's useful to specify that you don't want to maintain an ordering on data elements or data classes, but just always view them in alphabetical order.  
It would be nice to have a flag that could be set (ideally, by the users in the UI), to 'turn off' ordering.

**Comment by olliefreeman** [comment]
this should be a UI issue where just pass the query param "sort=label" that will sort the entries alphabetically. We should not be adding additional flags to do anything like this

**Comment by olliefreeman** [comment]
there is a piece of work on idx and siblings work which i need to do after #241 is sorted. Whilst i do this i will be adding an endpoint to "reset" all idx values for content inside an object.

**Comment by joe-crawford** [comment]
Would be desirable to allow treating certain specific ModelItem collections as unordered, as otherwise the system won't remember which should be displayed as ordered and the default ordering by index may be arbitrary (in insertion order if creating ModelItems by the UI).

Options to allow treating some ModelItem (element, class, etc.) collections as unordered:
1. Treat a collection of MIs all having index of 0 as unordered. When adding a new MI to an unordered collection, keep all the indices as 0. Add a method to the parent objects to remove ordering from a collection, e.g. `POST .../dataElements/unsort` or `POST .../dataElements/removeOrder`.
2. Add a flag to the parent objects to specify the default sort field for the collection, to allow storing a custom default sort field for each collection.
